src/wr_control_drive_arm/src/ArmControlActionServer.cpp

```diff
@@ -113,23 +117,37 @@ void execute(const control_msgs::FollowJointTrajectoryGoalConstPtr &goal,
             return;
         }
 
+        // Copy of the velocities received from MoveIt
         std::vector<double> velocityCopies{currTargetPosition.velocities};
 
         for (uint32_t i = 0; i < goal->trajectory.joint_names.size(); ++i) {
-            velocityCopies.at(i) *= abs(namedPositionMonitors.at(goal->trajectory.joint_names.at(i))->getCountsPerRotation());
+
+            // The joint that is currently being scaled
+            auto currJoint = goal->trajectory.joint_names.at(i);
+
+            // The position monitor whose velocity is currently being scaled
+            auto currPosMtr = namedPositionMonitors.at(currJoint);
```

Comment:
Avoid copying the `std::shared_ptr`:
```suggestion
            auto& currPosMtr = *namedPositionMonitors.at(currJoint);
```